packages/cdktf/lib/terraform-resource.ts

```diff
@@ -347,4 +381,84 @@ export class TerraformResource
   public addMoveTarget(moveTarget: string) {
     this._addResourceTarget(moveTarget);
   }
+
+  /**
+   * Moves this resource to the resource corresponding to "id"
+   * @param id Full id of resource to move to
+   */
+  public moveToId(id: string) {
+    if (this._movedById) {
+      throw new Error(`
+      ${this.node.id} has been given two separate moved operations.
+
+      {
+        from: ${this._movedById.from}
+        to: ${this._movedById.to}
+      }
+      {
+        from: ${id}
+        to: ${this.terraformResourceType}.${this.friendlyUniqueId} (Resource calling the move to operation)
+      }
+
+      Only one move operation can occur per plan/apply. Remove one of the operations.
+      `);
+    }
+    this._movedById = {
+      to: id,
+      from: `${this.terraformResourceType}.${this.friendlyUniqueId}`,
+    };
+    this._hasMoved = true;
+    this.node.addValidation(
+      new ValidateTerraformVersion(
+        ">=1.5",
+        `Resource move functionality is only supported for Terraform >=1.5. Please upgrade your Terraform version.`
+      )
+    );
+  }
+
+  /**
+   * Move the resource corresponding to "id" to this resource. Note that the resource being moved from must be marked as moved using it's instance function.
+   * @param id Full id of resource being moved from
+   */
+  public moveFromId(id: string) {
+    if (this._movedById) {
+      throw new Error(`
+      ${this.node.id} has been given two separate moved operations.
+
+      {
+        from: ${this._movedById.from}
+        to: ${this._movedById.to}
+      }
+      {
+        from: ${this.terraformResourceType}.${this.friendlyUniqueId} (Resource calling the move from operation)
+        to: ${id}
+      }
+
+      Only one move operation can occur plan/apply. Remove one of the operations.
+      `);
+    }
+    this._movedById = {
+      to: `${this.terraformResourceType}.${this.friendlyUniqueId}`,
+      from: id,
+    };
+    this.node.addValidation(
+      new ValidateTerraformVersion(
```

Comment:
I wonder if a single instance of this specific validation as e.g. moveBlockValidation would be possible, DRYing this up so we can keep the messaging concise :)

Reply:
Widdled it down to just two instances of the validation– seems that taking it down to one would require some more serious refactoring, at least if the validation were to occur in the `TerraformResource` class. Going to see if the validation can instead take place on the level of the stack to avoid the potential onslaught of errors.

Reply:
Was able to put the validation as a part of the synth process.

In `synthesize` of `StackSynthesizer`:

```typescript
if (this.stack.hasResourceMove()) {
      this.stack.node.addValidation(
        new ValidateTerraformVersion(
          ">=1.5",
          `Resource move functionality is only supported for Terraform >=1.5. Please upgrade your Terraform version.`
        )
      );
    }
```

Then in `TerraformStack` we just go through the elements of the stack checking if some resource of the stack has a resource move:

```typescript
public hasResourceMove(): boolean {
    return terraformElements(this).some((e) => {
      if (TerraformResource.isTerraformResource(e) && e.hasResourceMove()) {
        return true
      }
      return false;
    })
  }
```

**Andddd it works as expected**

_in the aws-move example_: Using `TERRAFORM_BINARY_NAME=../../../../terraform_1.4.7 STEP_2=1 cdktfl synth`

we get the following failure:
<img width="894" alt="Screenshot 2023-11-17 at 12 02 21 PM" src="https://github.com/hashicorp/terraform-cdk/assets/72527044/da1a1880-0816-4f30-8e5a-241ed4adcfe2">

Could be a point of improvement for import as well in a follow-up PR

Reply:
Jut to be clear: I mean put `const moveValidation = new ValidateTerraformVersion(...)` on the top of the file and use ` this.stack.node.addValidation(moveValidation)` in the places you have it right now. But yeah nothing urgent

Reply:
Ah I understand now. Thought you were referring to the issue where the validation error pops up a bunch of times in the case where there are a bunch of move statements.